modules/agar/src/main/ts/ephox/agar/api/Clipboard.ts

```diff
@@ -8,31 +8,39 @@ import * as ChainSequence from './ChainSequence';
 import { Step } from './Step';
 import { cFindIn } from './UiFinder';
 
-const cPasteDataTransfer = (mutator: (dataTransfer: DataTransfer) => void): Chain<SugarElement<any>, SugarElement<any>> =>
-  Chain.op<SugarElement<any>>((target) => {
-    const win = getWindowFromElement(target);
-    const dataTransfer = createDataTransfer();
-    const event = createPasteEvent(win, 0, 0, dataTransfer);
+const pasteDataTransfer = (target: SugarElement<Element>, mutator: (dataTransfer: DataTransfer) => void): void => {
```

Comment:
I wondered with these new APIs if we should instead make `target` optional and have it default to the document body. I didn't do that though as I figured in most cases that wouldn't make sense.